lasagne/tests/layers/test_recurrent.py

```diff
@@ -189,6 +189,34 @@ def test_custom_recurrent_arbitrary_shape():
     assert out_shape == (n_batch, n_steps, n_out_filters, width, height)
 
 
+def test_custom_recurrent_arbitrary_shape_with_mask():
+    # Check that the custom recurrent layer can handle more than 1 feature dim
+    # when masking the input
+    n_batch, n_steps, n_channels, width, height = (2, 3, 4, 5, 6)
+    n_out_filters = 7
+    filter_shape = (3, 3)
+    l_in = lasagne.layers.InputLayer(
+        (n_batch, n_steps, n_channels, width, height))
+    l_mask = lasagne.layers.InputLayer((n_batch, n_steps))
+    l_in_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((None, n_channels, width, height)),
+        n_out_filters, filter_shape, pad='same')
+    l_hid_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((None, n_out_filters, width, height)),
+        n_out_filters, filter_shape, pad='same')
```

Comment:
Could you modify this test such that the hidden dimensionality is different from the input dimensionality? This should uncover that your current fix of the mask dimensionality is wrong. I'm not sure there's a sensible use case for this -- maybe just reshape l_in_to_hid to 5 dimensions and then use a DenseLayer(..., num_leading_dims=-1) for hid_to_hid so the dimensionality stays at 5.